File: ios/CodersCamera.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CodersCamera, NSObject)

RCT_EXTERN_METHOD(openCamera:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/CodersCamera.swift
@objc(CodersCamera)
class CodersCamera: NSObject {

    @objc(openCamera:withRejecter:)
    func openCamera(resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        let imagePicker = UIImagePickerController();
        imagePicker.sourceType = .photoLibrary;

        DispatchQueue.main.async {
          RCTPresentedViewController()?.present(imagePicker, animated: true);
        }

        resolve("Camera aberta com sucesso!")
    }
}
